### PR TITLE
Introduce CM_MongoDb_Client::rename()

### PR DESCRIPTION
**Pull request by zazabe**

### CODE REVIEW

**Comment by zazabe** [comment]
@fvovan please review

**Comment by zazabe** [comment]
FYI:

> WARNING
If the renameCollection operation does not complete, the target collection and indexes will not be usable and will require manual intervention to clean up.
WARNING
This command obtains a global write lock and will block other operations until it has completed.

https://docs.mongodb.com/manual/reference/command/renameCollection/#behavior

**Comment by zazabe** [comment]
cc @alexispeter @njam

File: library/CM/MongoDb/Client.php [hunk 1]
```diff
@@ -243,6 +243,36 @@ public function count($collection, array $criteria = null, array $aggregation =
...
+        if (!$dropTarget && $this->existsCollection($collectionTarget)) {
+            throw new CM_MongoDb_Exception('Target collection already exists', null, [
+                'collectionSource' => $collectionSource,
+                'collectionTarget' => $collectionTarget
+            ]);
+        }
+        $result = $this->_getClient()->selectDB('admin')->command([
+            'renameCollection' => $this->_getDatabaseName() . '.' . $collectionSource,
+            'to'               => $this->_getDatabaseName() . '.' . $collectionTarget,
+            'dropTarget'       => $dropTarget ? 'true' : 'false'
```
Comment (fvovan) [inline library/CM/MongoDb/Client.php]: It's not necessary to pass stringified values there. just boolean `$dropTarget` will be okay.

File: library/CM/MongoDb/Client.php [hunk 2]
```diff
@@ -243,6 +243,36 @@ public function count($collection, array $criteria = null, array $aggregation =
...
     /**
+     * @param string    $collectionSource
+     * @param string    $collectionTarget
+     * @param bool|null $dropTarget
+     * @return array
+     * @throws CM_MongoDb_Exception
+     */
+    public function rename($collectionSource, $collectionTarget, $dropTarget = null) {
+        $dropTarget = (bool) $dropTarget;
+        if (!$this->existsCollection($collectionSource)) {
```
Comment (fvovan) [inline library/CM/MongoDb/Client.php]: It would be good to explicetely cast `$collectionSource` and `$collectionTarget` to strings

**Comment by zazabe** [comment]
@fvovan please check again

> Additionally we need to verify that mongo user has appropriate permissions on admin database on production

@njam wdyt?

**Comment by fvovan** [comment]
lgtm

**Comment by zazabe** [comment]
@njam should we drop this? see `mongo-rename` conversation...

**Comment by njam** [comment]
Would be good to have in general.

@alexispeter does the behaviour change with the [new mongodb client](https://github.com/cargomedia/cm/pull/2331)? If it's similar then maybe keep it?

**Comment by alexispeter** [comment]
The new client doesn't have a renameCollection function either, so you would also have to run a command directly on the admin db.

**Comment by njam** [comment]
I understand it work in general, but not on production, probably due to permissions.
I would still keep it.
wdyt?

**Comment by alexispeter** [comment]
Can't hurt.

**Comment by zazabe** [comment]
ok, let's merge it then!

**Pull request merged**